struct `rustc_plugin::Registry` does not have a field named `mir_passes`

Clippy won't compile on the latest rust nightly (rustc 1.17.0-nightly (824c9ebbd 2017-03-12), with this error:
````
error[E0026]: struct `rustc_plugin::Registry` does not have a field named `mir_passes`
  --> /home/nicholas/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy-0.0.118/src/main.rs:92:60
   |
92 |                                                            mir_passes,
   |                                                            ^^^^^^^^^^ struct `rustc_plugin::Registry` does not have field `mir_passes`

error: aborting due to previous error

error: failed to compile `clippy v0.0.118`, intermediate artifacts can be found at `/tmp/cargo-install.smZuH7anGt1C`

Caused by:
  Could not compile `clippy`.

To learn more, run the command again with --verbose.
````
Thanks for your help 😄

--- a/clippy_lints/src/items_after_statements.rs
+++ b/clippy_lints/src/items_after_statements.rs
@@ -58,6 +58,10 @@ impl EarlyLintPass for ItemsAfterStatements {
                 if in_macro(cx, it.span) {
                     return;
                 }
+                if let ItemKind::MacroDef(..) = it.node {
+                    // do not lint `macro_rules`, but continue processing further statements
+                    continue;
+                }
                 span_lint(cx,
                           ITEMS_AFTER_STATEMENTS,
                           it.span,

--- a/src/main.rs
+++ b/src/main.rs
@@ -89,7 +89,6 @@ impl<'a> CompilerCalls<'a> for ClippyCompilerCalls {
                                                            lint_groups,
                                                            llvm_passes,
                                                            attributes,
-                                                           mir_passes,
                                                            .. } = registry;
                     let sess = &state.session;
                     let mut ls = sess.lint_store.borrow_mut();
@@ -105,7 +104,6 @@ impl<'a> CompilerCalls<'a> for ClippyCompilerCalls {
                     }
 
                     sess.plugin_llvm_passes.borrow_mut().extend(llvm_passes);
-                    sess.mir_passes.borrow_mut().extend(mir_passes);
                     sess.plugin_attributes.borrow_mut().extend(attributes);
                 }
                 old(state);

--- a/tests/ui/item_after_statement.rs
+++ b/tests/ui/item_after_statement.rs
@@ -17,3 +17,14 @@ fn main() {
     fn foo() { println!("foo"); }
     foo();
 }
+
+fn mac() {
+    let mut a = 5;
+    println!("{}", a);
+    // do not lint this, because it needs to be after `a`
+    macro_rules! b {
+        () => {{ a = 6 }}
+    }
+    b!();
+    println!("{}", a);
+}

